### PR TITLE
Implement cockpit view

### PR DESCRIPTION
**Pull request by dethrace-labs**
- Fixes cockpit camera direction and position
- Fixes skybox model and rendering used in cockpit mode
- Updates `BR_MODU_*` enum names from BRender 1.3 that seem to match better than the BRender 1.2 SDK names
- Added support for rendering with `BR_RSTYLE_EDGES` - useful for debugging
- Reverted to using the colour_buffer width/height/offset when reading the depth buffer which resolves 1 and 2 from #282

https://user-images.githubusercontent.com/78985374/218414238-e247dd6e-64f4-4412-889b-acc93c617ec6.mov

### CODE REVIEW

File: src/BRSRC13/CORE/V1DB/render.c [hunk 1]
```diff
@@ -265,7 +265,38 @@ void BrDbSceneRenderBegin(br_actor* world, br_actor* camera) {
...
+
+/*
+ * Collect transforms from camera to root
+ *
+ * Make a stack of cumulative transforms for each level between
+ * the camera and the root - this is so that model->view
+ * transforms can use the shortest route, rather than via the root
+ */
+#define MAX_CAMERA_DEPTH 16
+    for (i = 0; i < MAX_CAMERA_DEPTH; i++)
```
Comment (madebr) [inline src/BRSRC13/CORE/V1DB/render.c:280]:
```suggestion
    for (i = 0; i < BR_ASIZE(v1db.camera_path); i++)
```

File: src/BRSRC13/CORE/V1DB/render.c [hunk 2]
```diff
@@ -265,7 +265,38 @@ void BrDbSceneRenderBegin(br_actor* world, br_actor* camera) {
...
+        v1db.camera_path[i].a = NULL;
```
Comment (madebr) [inline src/BRSRC13/CORE/V1DB/render.c:281]:
```suggestion
    for (i = 0; i < BR_ASIZE(v1db.camera_path); i++) {
        v1db.camera_path[i].a = NULL;
    }
```

File: src/DETHRACE/constants.h [hunk 3]
```diff
@@ -493,7 +493,7 @@ enum {
 
 #define TIME_CONV_THING 0.0005f
 
-#define OPPONENT_COUNT 5
+#define OPPONENT_COUNT 0
```
Comment (madebr) [inline src/DETHRACE/constants.h:496]: Don't forget to undo this before merging.

File: src/harness/renderers/gl/gl_renderer.c [hunk 4]
```diff
@@ -29,13 +29,16 @@ static br_pixelmap *last_colour_buffer, *last_depth_buffer;
...
-static int generated_current_framebuffer_for_this_frame = 0;
 
 static br_pixelmap *last_colour_buffer, *last_depth_buffer;
 
 static tStored_material* current_material;
 static br_pixelmap* current_shade_table;
 
+// Increment flush_counter during flush, and upload_counter when we upload the framebuffer texture.
+// If the counters are equal, we can avoid re-uploading the same thing.
+static int flush_counter = 0, framebuffer_upload_counter = 0;
```
Comment (madebr) [inline src/harness/renderers/gl/gl_renderer.c:40]: I would make these unsigned. Overflow for signed types is UB, but not for unsigned ones.

Comment (dethrace-labs) [reply src/harness/renderers/gl/gl_renderer.c]: thanks, done 👍

**Pull request merged**